src/braket/aws/aws_device.py

```diff
@@ -304,7 +305,7 @@ def _default_max_parallel(self):
 
     @staticmethod
     def _aws_session_for_device(device_arn: str, aws_session: Optional[AwsSession]) -> AwsSession:
-        """AwsSession: Returns an AwsSession for the device ARN. """
+        """AwsSession: Returns an AwsSession for the device ARN."""
```

Comment:
Nope;  get_device will fail on a simulator that isn't available in the region

Reply:
Yes, that's true. My thought here was that we should try to define the semantics for our methods/fields and have them apply throughout the code. We are fixing a problem which arose because some of our assumptions no longer hold (a simulator is no longer available in all regions). Updating all the methods to reflect that would be helpful. It would become progressively harder to update these as more functionality is built on top of these methods, and we would have to resolve to adding in additional if conditions making the code hard to read and maintain. Documenting our assumptions here would also be helpful for someone who's looking at the code later and doesn't have context on which pieces changed in the past and why. For instance, has the DEVICE_REGIONS dict is based on provider name which doesn't appear to be granular enough for its usage now. Is it supposed to reflect the regions a provider has at least one device in now ?  Would it help to update the field and its usage as such ?